x-pack/plugins/lens/public/xy_visualization/expression.tsx

```diff
@@ -561,17 +561,13 @@ export function XYChart({
       });
     }
 
-    const xAxisFieldName = table.columns.find((el) => el.id === layer.xAccessor)?.meta?.field;
-    const timeFieldName = xDomain && xAxisFieldName;
```

Comment:
This prevents clicking a bar in a time based x domain from updating the time range.
We should not simply remove this code, but check whether the underlying x domain is a date or a number - in case of a date, pass the time field name, in case of a number do not pass it.